src/lib/x509/certstor_system_windows/certstor_windows.cpp

```diff
@@ -185,6 +188,52 @@ std::vector<X509_Certificate> find_cert_by_dn_and_key_id(const Botan::X509_DN& s
 
    return search_cert_stores(blob, find_type, filter, return_on_first_found);
    }
+
+/** Handle certificates that do not adhere to RFC 3280 using a subject key identifier
+ *  that is not equal to the SHA-1 of the public key (w/o algorithm identifier)
+ *
+ *  This method lazily builds a cache of certificates found in previous queries as well
+ *  as negative results for @p key_hash queries that didn't find a certificate.
+ *
+ *  See here for further details: https://github.com/randombit/botan/issues/2779
+ */
+std::optional<X509_Certificate> find_cert_by_pubkey_sha1_via_exhaustive_search(const std::vector<uint8_t> &key_hash)
+   {
+   static std::map<std::vector<uint8_t>, std::optional<X509_Certificate>> non_rfc3289_certs;
+
+   if(const auto cache_hit = non_rfc3289_certs.find(key_hash); cache_hit != non_rfc3289_certs.end())
+      {
+      return cache_hit->second;
+      }
+
+   auto sha1 = HashFunction::create("SHA-1");
```

Comment:
Should use `create_or_throw` here - it shouldn't ever fail especially as you added SHA-1 as an explicit dependency, but if it did it would return null and then we would null-deref instead of throwing.